python: catch more DBus exceptions

Hjdskes opened the pull request:
`Gio.bus_get_sync` may return `None` when the DBus service is not available (see libratbag/piper#2). In addition to this, the service may time out when it is being started in which case `Gio.DBusProxy.new_sync` will raise a `GLib.Gerror`.

This commit checks for both scenarios and raises a `RatbagdDBusUnavailable`
exception when either one of them occurs. This will in turn be caught in for
example Piper to properly deal with this. This fixes libratbag/piper#2.

@whot  I hope this is acceptable as my mandatory GSoC contribution to Piper!

Hjdskes committed:
python: catch more DBus exceptions

Gio.bus_get_sync may return None when the DBus service is not available (see
libratbag/piper#2). In addition to this, the service may time out when it is
being started in which case Gio.DBusProxy.new_sync will raise a GLib.Gerror.

This commit checks for both scenarios and raises a RatbagdDBusUnavailable
exception when either one of them occurs. This will in turn be caught in for
example Piper to properly deal with this.

whot commented on python/ratbagd/__init__.py:
```diff
@@ -35,13 +35,20 @@ class RatbagdDBusUnavailable(BaseException):
-                                             object_path,
-                                             'org.freedesktop.ratbag1.{}'.format(interface),
-                                             None)
+        if self._dbus == None:
```
I learned this the hard way: "if foo == None" is not correct, we should use "if foo is None". Forgot what the exact difference is (Google remembers :) but it matters enough to be annoying.

Hjdskes replied:
Right, `is` checks for address equivalence and `==` checks for value equivalence.

whot reviewed (changes requested):
just a tiny change, I wish I could do these things directly in the interface here...

Thanks for the patch though, please force-push and ping me in a comment on the PR so I can merge it asap

Hjdskes committed:
python: replace == None with is None

== checks for value equivalence, but in this case what we
really want is address equivalence. Hence we use is.

Hjdskes commented:
@whot Ping! I also replaced another `== None` instance in a separate commit. I can squash those if you prefer, but you can also do this from the GitHub interface when you merge the PR.
